src/index.js

```diff
@@ -65,7 +67,8 @@ export default function createCopyOnWriteState<T>(baseState: T) {
         `the returned Provider, and/or delay your mutate calls until the component ` +
         `tree is moutned.`
     );
-    const nextState = produce(currentState, fn);
+    const fnWithCurrenState = (draft) => fn(draft, ((currentState: any): $ReadOnly<T>));
+    const nextState = produce(currentState, fnWithCurrenState);
```

Comment:
Can we just inline `fnWithCurrenState`?

```js
const nextState = produce(currentState, draft => fn(draft, currentState));
```

Reply:
yep, I initially wanted to add flow ignore on currentState because of the following:
```
Cannot call fn with currentState bound to state because inexact T [1] is incompatible with exact $ReadOnly [2].

 [2] 23│ type MutateFn<T> = (draft: T, state: $ReadOnly<T>) => void;
       :
 [1] 56│   let currentState: T = baseState;
       :
     68│     );
     69│     const nextState = produce(
     70│       currentState,
     71│       draft => fn(draft, currentState),
     72│     );
     73│     if (nextState !== currentState) {
     74│       currentState = nextState;

```

Reply:
I just trashed flow for now, so it doesn't matter anymore :D 

https://github.com/aweary/react-copy-write/pull/33